### PR TITLE
feat(clerk-js,types): Introduce UserOrganizationInvitation

### PR DESCRIPTION
**Pull request by panteliselef**
+ UserOrganizationInvitationResource
+ UserOrganizationInvitationJSON
+ ClerkPaginatedResponse that handles FAPI paginated responses

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

### CODE REVIEW

**Review by anagstef — APPROVED** [review]

File: packages/clerk-js/src/core/resources/User.ts [hunk 1]
```diff
@@ -250,6 +252,10 @@ export class User extends BaseResource implements UserResource {
     });
   };
 
+  getOrganizationInvitations = (params?: GetUserOrganizationInvitations) => {
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/core/resources/User.ts:255]: I think we should either describe the whole method with a type (`getOrganizationInvitations` -> `type GetOrganizationInvitations`) or just the params (`params` -> `type GetUserOrganizationInvitationsParams`)

If unsure, let's start by typing the params. We can reuse the type if needed to create `GetOrganizationInvitations` in the future. 

```suggestion
  getOrganizationInvitations = (params?: GetUserOrganizationInvitationsParams) => {
```

File: packages/types/src/user.ts [hunk 2]
```diff
@@ -93,6 +95,9 @@ export interface UserResource extends ClerkResource {
   getSessions: () => Promise<SessionWithActivitiesResource[]>;
   setProfileImage: (params: SetProfileImageParams) => Promise<ImageResource>;
   createExternalAccount: (params: CreateExternalAccountParams) => Promise<ExternalAccountResource>;
+  getOrganizationInvitations: (
+    params?: GetUserOrganizationInvitations,
```
Comment (nikosdouvlis) [inline packages/types/src/user.ts:99]: Same

File: packages/types/src/userOrganizationInvitation.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import type { MembershipRole } from './organizationMembership';
+import type { ClerkResource } from './resource';
+
+declare global {
+  /**
+   * If you want to provide custom types for the organizationInvitation.publicMetadata
+   * object, simply redeclare this rule in the global namespace.
+   * Every organizationInvitation object will use the provided type.
+   */
+  interface UserOrganizationInvitationPublicMetadata {
```
Comment (nikosdouvlis) [inline packages/types/src/userOrganizationInvitation.ts:12]: ✔️

**Review by nikosdouvlis — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**